### PR TITLE
Fix benign error toast for public annotations and other improvements

### PR DESCRIPTION
**Pull request by philippotto**
- Don't show error toast when fetching token or teams failed
- Fix padding of share button
- Make some modals lazy

### URL of deployed dev instance (used for testing):
- https://fixtoast.webknossos.xyz

### Steps to test:
- create a public sharing link for an annotation and open that in incognito tab --> no error toast should be visible
- the share button shouldn't look off, anymore

### Issues:
- fixes https://github.com/scalableminds/webknossos/issues/6262

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/admin/admin_rest_api.ts [hunk 1]
```diff
@@ -339,13 +339,17 @@ export function updateTaskType(taskTypeId: string, taskType: APITaskType): Promi
 
 // ### Teams
 export async function getTeams(): Promise<Array<APITeam>> {
-  const teams = await Request.receiveJSON("/api/teams");
+  const teams = await Request.receiveJSON("/api/teams", {
+    doNotInvestigate: true,
```
Comment (daniel-wer) [inline frontend/javascripts/admin/admin_rest_api.ts:343]: In `share_modal_view.tsx`, `showErrorToast: false` is used twice for this use case. However, I think your solution makes more sense, maybe you can unify the behavior for the share_model code as well :)

File: frontend/javascripts/libs/react_helpers.tsx [hunk 2]
```diff
@@ -67,4 +67,22 @@ export function usePolledState(callback: (arg0: OxalisState) => void, interval:
     callback(state);
   }, interval);
 }
+
+export function makeModalLazy<T extends { isVisible: boolean }>(
```
Comment (daniel-wer) [inline frontend/javascripts/libs/react_helpers.tsx:71]: There's no reason this wouldn't work for any React element, right? Maybe rename to `makeComponentLazy`?

**Review by daniel-wer — APPROVED** [review]
Nice fixes, works as advertised! I made two suggestions, but nothing blocking :+1:

File: frontend/javascripts/libs/react_helpers.tsx (hunk 2 above)
Comment (philippotto) [reply frontend/javascripts/libs/react_helpers.tsx:71]: The component needs to have a `isVisible` prop, but you are right. It's not special to modals. I'll rename it :)

File: frontend/javascripts/admin/admin_rest_api.ts (hunk 1 above)
Comment (philippotto) [reply frontend/javascripts/admin/admin_rest_api.ts:343]: To follow up with our in-person convo: I could remove the request parameters from `share_modal_view.tsx` by checking whether the user is logged in. However, the `doNotInvestigate` params in this module are quite handy for the team related selection UIs (`PermissionsAndTeamsModalView` and `TeamSelectionComponent`) which I didn't want to adapt for this hot-fix PR.

**Pull request merged**